### PR TITLE
Feature: Scripts for building and publishing

### PR DESCRIPTION
**Pull request by veskil**
This PR introduces both a Github Actions job and a shell script that can be used to tag and push releases, and then build and upload Neffint to PyPI

The shell script is taken from here: https://github.com/xsuite/xsuite/blob/main/release.sh

The Github Actions job is largely based on the following: https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/

### CODE REVIEW

**Comment by veskil** [comment]
@giadarol since you expressed an interest in seeing how publishing to PyPI can be done with github actions, here is the solution I came up with. It is mostly based on [1], but updated to use OIDC authentication instead of an API token, since that was recommended by [2] (the publishers of the main uploading script used in the job), since it is apparently the current best practice.

See commit 97e58b213ba55891f556949ab0b4dc206fcbdab7 for the older (API token) approach, which is used by [1].

Additional setup required:
- Using the API token approach: Copy an API key from PyPI into the repository secrets on Github.
- Using the OIDC approach: Add the repository name and organization as a Trusted Publisher on the package settings on PyPI 

[1]  https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/
[2] https://github.com/pypa/gh-action-pypi-publish#specifying-a-different-username

File: setup.py [hunk 1]
```diff
@@ -1,30 +1,37 @@
...
 root_dir = Path(__file__).parent.absolute()
+
+# Read README.md to use as long description in package metadata
 with (root_dir / "README.md").open("rt") as readme_file:
     long_description = readme_file.read().strip()
 
+# Read semantic version number from _version.py
+version_file_namespace = {} # type:ignore
+with (root_dir / "neffint" / "_version.py").open("r") as version_file:
+    exec(version_file.read(), version_file_namespace)
```
Comment (nmounet) [inline setup.py:26]: Isn't there another (more pythonic...) way to do this? What about importing the version number, simply?

File: setup.py [hunk 2]
```diff
@@ -1,30 +1,37 @@
...
+version = version_file_namespace["__version__"]
+
 setup(
     name="neffint",
-    version=VERSION,
-    description="A python package for computing Fourier integrals using a Filon type method with non-equidistant grid spacing.",
+    version=version,
+    description="Computing Fourier integrals using a Filon type method with non-equidistant grid spacing.",
```
Comment (nmounet) [inline setup.py:32]: I have to say I prefer the initial version of `description`... or is it that `A python package` is anyway printed by PyPI right before its description, or something like this?

**Review by nmounet — COMMENTED** [review]
looks good to me. I have a few questions but they might be irrelevant.

File: setup.py (hunk 1 above)
Comment (nmounet) [reply setup.py:26]: what about something like (from [here](https://packaging.python.org/en/latest/guides/single-sourcing-package-version/) )

```
def get_version(rel_path):
    for line in read(rel_path).splitlines():
        if line.startswith('__version__'):
            delim = '"' if '"' in line else "'"
            return line.split(delim)[1]
    else:
        raise RuntimeError("Unable to find version string.")
```

and then using it with `version = get_version(os.path.join("neffint","_version.py"))`?

**Comment by veskil** [comment]
I agree with both suggestions

**Review by nmounet — APPROVED** [review]

**Pull request merged**